docs/azure-tre-overview/cost-reporting.md

```diff
@@ -0,0 +1,295 @@
+          }]
+        }]
+}
+```
+
+## Sequence Diagram
+
+[source](https://swimlanes.io/u/JlMQKbGqd?rev=4)
+
+[![Sequence Diagram](../assets/cost-reporting-sequence-diagram.png)](../assets/cost-reporting-sequence-diagram.png)
+
+
+## Limitations and notes
+
+* Cost and usage data is typically available in Cost Management within 8-24 hours.
+
+* Tags aren't applied to historical data, template authors need to make sure all relavent [Azure resources of a TRE resource are tagged as instructed](#azure-resources-tagging).
+
+* Cost records might include [multiple currencies](https://azure.microsoft.com/en-us/blog/azure-cost-management-updates-july-2019/#currency) for the same date and TRE resource.
+
+* Once cost data becomes available in Cost Management, it will be retained for at least seven years. Only the last 13 months is available from the TRE Cost API and Azure Portal. For historical data before 13 months, please use [Exports](https://docs.microsoft.com/en-us/azure/cost-management-billing/costs/tutorial-export-acm-data?tabs=azure-portal) or the [UsageDetails API](https://docs.microsoft.com/en-us/rest/api/consumption/usage-details/list?tabs=HTTP).
+
+* For more information please refer to [Azure Cost Management](https://docs.microsoft.com/en-us/azure/cost-management-billing/costs/understand-cost-mgt-data) and Cost API swagger docs.
+
+
+## Azure Resources Tagging
+
+TRE Cost Reporting is based on Azure tagging to be able to generate cost report for core services, shared services, workspace, workspace services and user resources.
+Templates authors need to make sure that underling Azure resources are tagged with the following tags:
+
+
+| <div style="width:160px">Tag</div> | Value | Applies to |
+| ---------------------------------- | ----- | ---------- |
+| `tre_id` | Unique ID of the TRE instance | All resources of a TRE instance |
+| `tre_core_service_id` | Unique ID of the TRE instance | All TRE core azure resources |
+| `shared_service_id` | The shared service unique ID | Shared Services |
+| `workspace_id` | The workspace unique ID | Workspaces, Workspace Services and User Resources |
+| `workspace_service_id` | The workspace service unique ID | Workspace Services and User Resources |
+| `user_resource_id` | The user resoruce unique ID | User Resources |
+
+!!! Notes
+    Main Azure Container Registry and Storage Account are not be tagged as those resources are used to spin up more than one Azure TRE Instance.
+
+[![Resource tagging example](../assets/resource-tagging-example.png)](../assets/resource-tagging-example.png)
+
+
+## Cost management query API
+
+Cost management query API API, which Azure TRE Cost APIs are based upon, returns a flat result of all cost with every tag combination which exists on the filtered resources by the provided tag, meaning that a resource which has tre_id, tre_core_service_id, tre_workspace_id, tre_workspace_service_id, and tre_user_resource_id will get be summarized by all those tags.
```

Comment:
```suggestion
Cost management queries each API, which Azure TRE Cost APIs are based upon, returns a flat result of all cost with every tag combination which exists on the filtered resources by the provided tag, meaning that a resource which has tre_id, tre_core_service_id, tre_workspace_id, tre_workspace_service_id, and tre_user_resource_id will get be summarized by all those tags.
```

Reply:
```Cost management query API```  is the name of the rest api, I'll remove the double API, its a typo.